thunder/executors/torchex.py

```diff
@@ -1031,8 +1032,15 @@ def _addcmul_transform(a: TensorLike, b: TensorLike, c: TensorLike, /, *, value:
     return addcmul(a, b, c, value=value)
 
 
+def _lerp_checker(start: TensorLike, end: TensorLike, weight: Number | TensorLike) -> TensorLike:
+    if not isinstance(weight, (Number, TensorLike)):
```

Comment:
These two conditionals could be combined into a single line, but this is fine as written

Reply:
why do we not support NumberProxy here any more?

Reply:
Thank you, that was a mistake. Maybe we should add `NumberProxy` to `weight`'s type annotation, but this is apparently not the case for many other functions.